src/main/java/kitchenpos/table/TableService.java

```diff
@@ -0,0 +1,56 @@
+package kitchenpos.table;
+
+import kitchenpos.order.domain.OrderRepository;
+import kitchenpos.order.domain.OrderStatus;
+import kitchenpos.table.dto.*;
+import org.mapstruct.factory.Mappers;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class TableService {
+    private final OrderRepository orderRepository;
+    private final OrderTableRepository orderTableRepository;
+    private final OrderTableMapper mapper = Mappers.getMapper(OrderTableMapper.class);
+
+    public TableService(final OrderRepository orderRepository, final OrderTableRepository orderTableRepository) {
+        this.orderRepository = orderRepository;
+        this.orderTableRepository = orderTableRepository;
+    }
+
+    @Transactional
+    public OrderTableResponse create(final TableAddRequest orderTable) {
+        return mapper.toResponse(orderTableRepository.save(new OrderTable()));
+    }
+
+    public List<OrderTableResponse> list() {
+        return orderTableRepository.findAll()
+                .stream()
+                .map(mapper::toResponse)
+                .collect(Collectors.toList());
+    }
+
+    @Transactional
+    public OrderTableResponse changeEmpty(final Long orderTableId, final TableEmptyChangeRequest request) {
+        final OrderTable savedOrderTable = orderTableRepository.findById(orderTableId)
+                .orElseThrow(IllegalArgumentException::new);
+        if (orderRepository.existsByOrderTable_IdAndOrderStatusIn(orderTableId, Arrays.asList(OrderStatus.COOKING, OrderStatus.MEAL))) {
+            throw new IllegalArgumentException();
+        }
+        savedOrderTable.changeEmpty(request.isEmpty());
+        return mapper.toResponse(orderTableRepository.save(savedOrderTable));
+    }
+
+    @Transactional
+    public OrderTableResponse changeNumberOfGuests(final Long orderTableId, final TableNumberChangeRequest request) {
+        final int numberOfGuests = request.getNumberOfGuests();
+        final OrderTable savedOrderTable = orderTableRepository.findById(orderTableId)
+                .orElseThrow(IllegalArgumentException::new);
+        savedOrderTable.changeNumberOfGuests(numberOfGuests);
```

Comment:
지역 변수는 사용하기 직전에 선언해 코드가 적용되는 범위를 최대한 줄이는 것이 가독성을 향상시킬 수 있습니다.

```suggestion
        final OrderTable savedOrderTable = orderTableRepository.findById(orderTableId)
                .orElseThrow(IllegalArgumentException::new);
        final int numberOfGuests = request.getNumberOfGuests();
        savedOrderTable.changeNumberOfGuests(numberOfGuests);
```

개인적으론 이정돈 지역변수를 선언해서 의도를 나타내지 않아도 의도가 메서드를 통해 드러나기 때문에 생략해도 무방하다고 생각합니다.

```suggestion
        final OrderTable savedOrderTable = orderTableRepository.findById(orderTableId)
                .orElseThrow(IllegalArgumentException::new);
        savedOrderTable.changeNumberOfGuests(request.getNumberOfGuests(););
```